projects/dotcms-field-elements/src/components/dot-key-value/dot-key-value.tsx

```diff
@@ -0,0 +1,207 @@
+import {
+    Component,
+    Prop,
+    State,
+    Element,
+    Event,
+    EventEmitter,
+    Method,
+    Listen
+} from '@stencil/core';
+import Fragment from 'stencil-fragment';
+import {
+    DotFieldStatus,
+    DotFieldValueEvent,
+    DotFieldStatusEvent,
+    DotLabel,
+    KeyValue
+} from '../../models';
+import {
+    getClassNames,
+    getOriginalStatus,
+    getTagHint,
+    getTagError,
+    getTagLabel,
+    getErrorClass,
+    updateStatus
+} from '../../utils';
+
+@Component({
+    tag: 'dot-key-value',
+    styleUrl: 'dot-key-value.scss'
+})
+export class DotKeyValueComponent {
+    @Element() el: HTMLElement;
+    @Prop({ mutable: true }) value: string;
+    @Prop() name: string;
+    @Prop() fieldType: string;
+    @Prop() label: string;
+    @Prop() hint: string;
+    @Prop() keyPlaceholder: string;
+    @Prop() valuePlaceholder: string;
+    @Prop() required: boolean;
+    @Prop() requiredmessage: string;
+    @Prop() saveBtnLabel = 'Add';
+    @Prop() disabled = false;
+
+    @State() status: DotFieldStatus;
+    @State() keyValues: KeyValue[] = [];
+
+    @Event() valueChange: EventEmitter<DotFieldValueEvent>;
+    @Event() statusChange: EventEmitter<DotFieldStatusEvent>;
+
+    fieldInput: KeyValue = { key: '', value: '' };
+
+    /**
+     * Reset properties of the filed, clear value and emit events.
+     */
+    @Method()
+    reset(): void {
+        this.fieldInput = { key: '', value: '' };
+        this.keyValues = [];
+        this.status = getOriginalStatus(this.isValid());
+        this.emitStatusChange();
+        this.emitValueChange();
+    }
+
+    @Listen('deleteKeyValue')
```

Comment:
is redundant to do call this `deleteKeyValue` since we know is a key/value field, `deleteItem`?